### ISSUE TITLE
mismatch return type between Jedis and JedisCluster

### ISSUE THREAD

**Issue by bironran** [redis/jedis#2497]
Jedis 3.5.2 has the following signatures which are incompatible between `Jedis` and `JedisCluster`:

```
class Jedis {
        long xgroupDestroy(final String key, final String groupname);
        long xdel(final String key, final StreamEntryID... ids);
        long xtrim(final String key, final long maxLen, final boolean approximateLength);
        long xack(final String key, final String group, final StreamEntryID... ids);
}
```
```
class JedisCluster {
        Long xtrim(final String key, final long maxLen, final boolean approximateLength);
        Long xdel( String key, StreamEntryID... ids);
        Long xgroupDestroy(final String key, final String groupname);
        Long xack(final String key, final String group, final StreamEntryID... ids);
}
```

These have different (and incompatible) return type requiring and adapter class for a client that want to use JedisCluster or Jedis interchangeably.

**Comment by sazzad16** [comment]
@bironran Thank you for notifying us. This is a known issue to us but resolving this issue is not completely backward incompatible. As we don't have an exact way to judge the affect of resolution compared to irresolution, we are more likely to resolve this in upcoming major release. In the mean time, please use a workaround. Sorry for the inconvenience and thank you for having patience and being with us.

**Comment by sazzad16** [comment]
Resolved by #2533
